server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -329,12 +329,24 @@ public void testRegexValidator() throws Exception {
         String expectedRegex = "\\d+";
         Pattern expectedPattern = Pattern.compile(expectedRegex);
         RegexValidator regexValidator = new RegexValidator(expectedRegex);
+        RegexValidator regexValidatorMatcherFalse = new RegexValidator(expectedRegex, false);
+        RegexValidator regexValidatorMatcherTrue = new RegexValidator(expectedRegex, true);
 
         // Test that the pattern is correctly initialized
         assertNotNull(expectedPattern);
         assertNotNull(regexValidator.getPattern());
         assertEquals(expectedPattern.pattern(), regexValidator.getPattern().pattern());
 
+        // Test that checks the pattern and isMatching with the set value false parameters are working correctly during initialization
+        assertNotNull(regexValidatorMatcherFalse);
+        assertNotNull(regexValidatorMatcherFalse.getPattern());
+        assertEquals(expectedPattern.pattern(), regexValidatorMatcherFalse.getPattern().pattern());
+
+        // Test that checks the pattern and isMatching with the set value true parameters are working correctly during initialization
+        assertNotNull(regexValidatorMatcherTrue);
+        assertNotNull(regexValidatorMatcherTrue);
+        assertEquals(expectedPattern.pattern(), regexValidatorMatcherTrue.getPattern().pattern());
+
         // Test that validate() throws an exception for invalid input
         final RegexValidator finalValidator = new RegexValidator(expectedRegex);
```

Comment:
You need to make a copy of these tests (line 351-358) for `regexValidatorMatcherFalse`.  You should get the opposite behavior (exception when it matches and no exception when it doesn't match).